src/audio/dai-zephyr.c

```diff
@@ -534,7 +534,8 @@ static int dai_playback_params(struct comp_dev *dev, uint32_t period_bytes,
 		prev = dma_block_cfg;
 		prev->next_block = ++dma_block_cfg;
 	}
-	prev->next_block = dma_cfg->head_block;
+	if (prev && prev->next_block)
```

Comment:
is this correct? You overwrite `next_block` only if it isn't NULL? Before this patch it was set always

Reply:
hmm yeah doesn't make sense... will fix

Reply:
fixed